web-client/src/pages/session/test.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import axios from 'axios';
+
+import { User } from './types';
+import { Meta } from '../../layout/Meta';
+import { Main } from '../../templates/Main';
+import { AppConfig } from '../../utils/AppConfig';
+
+const LoginTest = () => {  
+  const userJson = localStorage.getItem(User.CURRENT);
+  const jwt = localStorage.getItem(User.JWT);
+
+  if (userJson == null || jwt == null) {
+    return (
+      <Main meta={<Meta title={AppConfig.title} description={AppConfig.description}/>}>
+        You are logged out
+      </Main>
+    )
+  }
+
+  try {
+    const user = JSON.parse(userJson);
+    (async () => await axios.get(`http://localhost:4000/v1/users/${user.id}`, {
+      headers: {
+        'Authorization': jwt
+      }
+    }));
```

Comment:
I don't think this is needed anymore. This seems like a function definition that doesn't get called. Nor do you need the result anyway.

Reply:
Also, I don't think React components can `await` inside them. Correct me if I'm wrong, but isn't the desired effect you want something like:

Step 1: Render a loading screen and start checking if you're logged in. This second bit requires some async stuff and might take a few seconds if the internet is slow.

Step 2: Depending on the response, you want to replace the loading screen with either "You are logged in as X" or "You are logged out".

If so may I suggest some code that roughly looks like this instead:

```
const LoginTest = () => {
  const getCurrentUser = async () => {
    // Do the thing that gets the JWT and makes a server thing.
    // return User if logged in, returns null if logged out
  }

  [status, setStatus] = useState('loading');
  [user, setUser] = useState(null);

  const checkIfLoggedInAndUpdateStatus = async () => {
    const user = await getCurrentUser();
    if (user != null) {
      setStatus('logged-in');
      setUser(user);
    } else {
      setStatus('logged-out');
    }
  }
  useAsyncEffect(checkIfLoggedInAndUpdateStatus);
  if (status == 'loading') {
    return <div>Loading</div>
  } else if (status == 'logged-in') {
    return <div>You are logged in as {user.email}</div>
  } else if (status == 'logged-out') {
    return <div>You are logged out</div>
  }
};

export default LoginTest;
```